src/logging.rs

```diff
@@ -1,53 +1,143 @@
 }
 
-/// Log a message to the Redis log with DEBUG log level.
-pub fn log_debug(message: &str) {
-    log(LogLevel::Debug, message);
+/// Log a message to Redis at the [RedisLogLevel::Debug] level.
+pub fn log_debug<T: AsRef<str>>(message: T) {
+    log(RedisLogLevel::Debug, message.as_ref());
 }
 
-/// Log a message to the Redis log with NOTICE log level.
-pub fn log_notice(message: &str) {
-    log(LogLevel::Notice, message);
+/// Log a message to Redis at the [RedisLogLevel::Notice] level.
+pub fn log_notice<T: AsRef<str>>(message: T) {
+    log(RedisLogLevel::Notice, message.as_ref());
 }
 
-/// Log a message to the Redis log with VERBOSE log level.
-pub fn log_verbose(message: &str) {
-    log(LogLevel::Verbose, message);
+/// Log a message to Redis at the [RedisLogLevel::Verbose] level.
+pub fn log_verbose<T: AsRef<str>>(message: T) {
+    log(RedisLogLevel::Verbose, message.as_ref());
 }
 
-/// Log a message to the Redis log with WARNING log level.
-pub fn log_warning(message: &str) {
-    log(LogLevel::Warning, message);
+/// Log a message to Redis at the [RedisLogLevel::Warning] level.
+pub fn log_warning<T: AsRef<str>>(message: T) {
+    log(RedisLogLevel::Warning, message.as_ref());
+}
+
+/// The [log] crate implementation of logging.
+pub mod standard_log_implementation {
+    use super::*;
+    use log::{Level, Metadata, Record, SetLoggerError};
+
+    static LOGGER: RedisGlobalLogger = RedisGlobalLogger;
+
+    /// The struct which has an implementation of the [log] crate's
+    /// logging interface.
+    ///
+    /// # Note
+    ///
+    /// Redis does not support logging at the [log::Level::Error] level,
+    /// so logging at this level will be converted to logging at the
+    /// [log::Level::Warn] level under the hood.
+    struct RedisGlobalLogger;
+
+    /// Sets this logger as a global logger. Use this method to set
+    /// up the logger. If this method is never called, the default
+    /// logger is used which redirects the logging to the standard
+    /// input/output streams.
+    ///
+    /// # Example
+    ///
+    /// This function may be called on a module startup, within the
+    /// module initialisation function (specified in the
+    /// [crate::redis_module] as the `init` argument, which will be used
+    /// for the module initialisation and will be passed to the
+    /// [raw::Export_RedisModule_Init] function when loading the
+    /// module).
+    #[allow(dead_code)]
+    pub fn setup() -> Result<(), SetLoggerError> {
+        log::set_logger(&LOGGER).map(|()| log::set_max_level(log::LevelFilter::Trace))
+    }
+
+    impl log::Log for RedisGlobalLogger {
+        fn enabled(&self, _: &Metadata) -> bool {
+            true
+        }
+
+        fn log(&self, record: &Record) {
+            if !self.enabled(record.metadata()) {
+                return;
+            }
+
+            let message = format!(
+                "'{}' {}:{}: {}",
+                record.module_path().unwrap_or_default(),
```

Comment:
If we will take the suggestion I made above then I guess we do not need to print the module here because Redis will do it for us.

Reply:
> The logging here may come from any crate, not just ours. If we use a dependency which also uses the log crate, this handler will be invoked, and so the module_path() will be that crate which logs

Ohh ok, in that case lets keep it.